### ISSUE TITLE
[SUGGESTION] Accept directories in 'Add new file'

### ISSUE THREAD

**Issue by sliekens** [microsoft/vscode-dotnettools#542]
### Describe the feature you'd like

In the file explorer, it's possible to specify a directory path when adding a new file.

![image](https://github.com/microsoft/vscode-dotnettools/assets/1583241/cf9e18fd-e586-4bee-92b8-685ad5d07937)

Missing directories are created as needed.

![image](https://github.com/microsoft/vscode-dotnettools/assets/1583241/9da8efdf-0671-4d00-a918-8769d7f2f327)

Similar functionality is currently not present in the Solution Explorer 'Add New File' dialog.

```
Starting Add a new file using template: Class...
Starting command: "dotnet" new class -n Some/NewLocation/Class1...
Failed at 'command: "dotnet" new class -n Some/NewLocation/Class1' (1769ms) with error: Failed with ExitCode(127)
Details
Name cannot contain any of the following characters '/' or character codes char(0)

For details on the exit code, refer to https://aka.ms/templating-exit-codes#127
```

### Environment Information

C# Dev Kit v0.4.10
VSCode 1.82.2
Windows 11 + WSL2

**Comment by smitpatel** [comment]
@sayedihashimi - Looking at add new item in VS, it doesn't accept directory. User can right click on solution explorer node where they want to add it or can move the file to right location after creating. Do we want to do anything different in C# dev kit?

**Comment by sayedihashimi** [comment]
> @sayedihashimi - Looking at add new item in VS, it doesn't accept directory. User can right click on solution explorer node where they want to add it or can move the file to right location after creating. Do we want to do anything different in C# dev kit?

It would be good to support both the right click gesture in Solution Explorer as well as the directory path since that is how VS Code normally behaves.

Regarding Visual Studio the new compact Add New Item dialog does respect directories if they are passed

![image](https://github.com/microsoft/vscode-dotnettools/assets/1283154/4e601a24-675f-4205-a3a3-39bfef08c46a)

This will create the Foo directory if it doesn't exist.

**Comment by smitpatel** [comment]
This is fixed in Sept release version 1.10.18

**Comment by sliekens** [comment]
Very cool, thank you.
